### ISSUE TITLE
Type mismatch for `highlightUntyped` VS Code extension setting prevents Sorbet client from starting

### ISSUE THREAD

**Issue by antn** [sorbet/sorbet#7578]
The changes in https://github.com/sorbet/sorbet/pull/7571 appear to have introduced this bug? cc @jez 

```
Unable to deserialize LSP request: Error deserializing JSON message: Expected field `SorbetInitializationOptions.highlightUntyped` to have value of type `boolean`, but had value `"nowhere"`.
```

![Image](https://github.com/sorbet/sorbet/assets/858504/e9bd5edd-1c88-478e-8133-027008e3e817)

Sorbet version: `Sorbet typechecker 0.5.11109 git 083179e6ee7c567296e36c002bdc8ca59d4b1cac debug_symbols=true clean=0`

VS Code extension version: `0.3.277`

**Comment by jez** [comment]
#7579

**Comment by dcecile** [comment]
I tried upgrading to `0.3.28`, but am still getting the error:

```
[Error - 2:23:56 PM] Starting client failed
  Message: Unable to deserialize LSP request: Error deserializing JSON message: Expected field `SorbetInitializationOptions.highlightUntyped` to have value of type `boolean`, but had value `"everywhere"`.
  Code: -32602 
libc++abi: Pure virtual function called!
/nix/store/1msbqh05wv7dgnppmr9qb4dbvnyiskwa-giza-env/lib/ruby/gems/3.2.0/gems/sorbet-0.5.11120/bin/srb: line 46: 31997 Abort trap: 6           "${sorbet}" "${args[@]}"
```

**Comment by jez** [comment]
As of 0.3.29, this should be completely fixed. It has the changes from #7571 but avoids the problem where sometimes the language server would fail to start.

For any further problems, feel free to open a new issue.

Closed via #7580
